cloudinit/config/schemas/schema-cloud-config-v1.json

```diff
@@ -34,10 +34,30 @@
           "type": "string"
         },
         "groups": {
-          "description": "Optional comma-separated string or list of groups to add the user to.",
-          "type": ["string", "array"],
-          "items": {"type": "string"},
-          "minItems": 1
+          "description": "Optional comma-separated string of groups to add the user to.",
+          "oneOf": [
+            {"type": "string"},
+            {
+              "type": "array",
+              "items": {
+                "type": ["string"]
+              },
+              "minItems": 1
+            },
+            {
+              "type": "object",
+              "patternProperties": {
+                "^.+$": {
+                  "label": "<group_name>",
+                  "description": "When providing an object for users.groups the ``<group_name>`` keys are the groups to add this user to",
+                  "deprecated": true,
```

Comment:
We do emit that warning. I have added tests to make sure it's covered:

* https://github.com/canonical/cloud-init/pull/1549/files#diff-665a8909889380de684d22ea92285daed075400bc82b20258487a15a12e420cfR384
* https://github.com/canonical/cloud-init/pull/1549/files#diff-ccf5aac2a6067d8a0782d74b172805f80e71225a212822e4f4e158f9ca1e6e3aR547

Reply:
> Something like <regex1> is deprecated if the given instance matches it?

More or less, yes.

My assumption was that when we encounter `"deprecated": true` in the schema, we'd find a way to log that without having to add separate logs to the code. That's obviously much harder to do in practice, and in cases like this, I'm not sure it even makes sense. I was more wondering if you were planning to try to handle cases like this. If not, that's totally fine, especially with how much extra work it might be to support such an obscure case.

Reply:
I think it is not that difficult. We already emit a warning even if the description is not given: https://github.com/canonical/cloud-init/pull/1549/files#diff-ccf5aac2a6067d8a0782d74b172805f80e71225a212822e4f4e158f9ca1e6e3aR555

We could easily extend the (given or not) description with the regex within [_validator_deprecated](_validator_deprecated) as that info is given in the `schema` argument.

Let me create a ticket.

Reply:
SC-1174